sphinxcontrib/httpdomain.py

```diff
@@ -25,19 +25,23 @@
 from sphinx.util import logging
 from sphinx.util.nodes import make_refnode
 from sphinx.util.docfields import GroupedField, TypedField
+from sphinx.util.docutils import LoggingReporter
 from sphinx.locale import _
 
 logger = logging.getLogger(__name__)
 
-# The env.get_doctree() lookup results in a pickle.load() call which is
-# expensive enough to dominate the runtime entirely when the number of endpoints
-# and references is large enough. The doctrees are generated during the read-
-# phase and we can cache their lookup during the write-phase significantly
-# improving performance.
-# Currently sphinxcontrib-httpdomain does not declare to support parallel read
-# support (parallel_read_safe is the default False) so we can simply use a
-# module global to hold the cache.
-_doctree_cache = {}
+
+class DummyDocument(object):
+    """Used where the signature requires a docutils.node.Document but only its reporter
+    is being used.
+
+    This helps avoiding the need to unpickle the document in some situations for
+    performance reasons. XXX: Is this really a performance problem (may be resolved
```

Comment:
Anecdotally, using even the latest version of Sphinx, not caching and re-loading doctrees from the pickle seems to cause a big slow-down during writing (my docset has 93 individual topic files). I didn't do any kind of rigorous timing or other perf measurement; just changed the code to not use the cache and build from scratch.